src/lib/logical_query_plan/lqp_translator.cpp

```diff
@@ -590,7 +590,22 @@ std::shared_ptr<AbstractExpression> LQPTranslator::_translate_expression(
       const auto subquery_expression = std::dynamic_pointer_cast<LQPSubqueryExpression>(expression);
       Assert(subquery_expression, "Expected LQPSubqueryExpression");
 
-      const auto subquery_pqp = translate_node(subquery_expression->lqp);
+      /**
+       * Notes on generating subquery PQPs:
+       *  a) For uncorrelated subqueries, this LQPTranslator's cache is used to create subquery PQPs. By using the
+       *     cache, operator results can be shared between identical parts in subquery PQPs and owning PQPs.
+       *
+       *  b) For correlated subqueries, a new LQPTranslator instance is used to create subquery PQPs.
+       *     In contrast to uncorrelated subqueries, operator results cannot be shared easily between subquery PQPs
+       *     and owning PQPs because correlated subqueries imply changing parameters on a row-by-row basis.
```

Comment:
Someone reading this might have the same question that I had in the previous discussion: What about uncorrelated parts of the subquery? Maybe explain that correlated subqueries are executed using `deep_copy`, which does not identify uncorrelated parts of the copied plan? As a result, those parts of the plan that would be deduplicated here would be duplicated, anyway, correct?

Reply:
Changed to 

```cpp
/**
 * Notes on generating subquery PQPs:
 *  a) For uncorrelated subqueries, operator results can be shared between identical parts in uncorrelated
 *     subqueries and outer queries. Therefore, this LQPTranslator instance is used to deduplicate subquery PQPs
 *     with _operator_by_lqp_node.
 *
 *  b) In contrast to uncorrelated subqueries, correlated subqueries cannot share identical parts with outer
 *     queries because ExpressionEvaluator::_evaluate_subquery_expression_for_row always deep-copies the whole PQP
 *     at evaluation time. The deep copy includes both correlated and uncorrelated parts.
 *     Consequently, a new LQPTranslator instance is used for correlated subqueries to prevent deduplication
 *     with outer queries.
 */
```

Reply:
The last sentence will leave people wondering why we do extra work to avoid the deduplication. Add a sentence about the consumer count and I'm happy.

Reply:
Added 

>  Besides, we also prevent correlated subqueries from increasing the consumer count of outer query operators, which would otherwise block the automatic clearing of results.

Reply:
"Besides"? Isn't that the whole reason why this PR exists?

This might seem nitpicky, but those are things that people might get confused about in a year from now.

Reply:
"This also prevents correlated subqueries ..." ?

Reply:
Haha, I was just about to complain about the "also". Sounds good. ;)